pkg/platform/baremetal/template.go

```diff
@@ -70,6 +70,23 @@ module "bare-metal-{{.ClusterName}}" {
 
   ignore_x509_cn_check   = {{.IgnoreX509CNCheck}}
   conntrack_max_per_core = {{.ConntrackMaxPerCore}}
+
+  {{- if .CLCSnippets}}
+  clc_snippets = {
+    {{- range $node_name, $clc_snippet_list := .CLCSnippets }}
+    "{{ $node_name}}" = [
+      {{- if $clc_snippet_list }}
```

Comment:
Rather than having this condition, I'd add a validation rule to ensure that we don't accept empty lists.